remote/save.go

```diff
@@ -35,11 +35,11 @@ func (i *Image) SaveAs(name string, additionalNames ...string) error {
 	if err != nil {
 		return errors.Wrap(err, "get image layers")
 	}
-	cfg.History = make([]v1.History, len(layers))
+	if len(cfg.History) != len(layers) {
+		cfg.History = make([]v1.History, len(layers))
+	}
```

Comment:
We shouldn't need this as we've been careful to update the history when instantiating an image and adding layers, but you never know...